### PR TITLE
Test/overview liked screen

### PR DESCRIPTION
**Pull request by ThomasPFMuller**
Added tests for LikedScreen and OverviewScreen, which also run the SeachBar and CategorySelector Composable, testing them through their interaction with the aforementioned screens. Will need to add a test for the price and time sliders to get the full functionality tested.

Test coverage in list_itineraries folder is now 87% (was ~66%).

image

### CODE REVIEW

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 1]
```diff
@@ -112,9 +81,91 @@ class LikedScreenTest {
...
+
+    // choose price range
+    var priceRange = 0f..100f
+    every { sliderPositionPriceState.value } returns priceRange
+
+    // SF and Switzerland itineraries should be displayed
+    testExpectedItinerariesDisplayed(listOf(FakeItinerary.SWITZERLAND, FakeItinerary.SAN_FRANCISCO))
+
+    priceRange = 40f..100f
+    every { sliderPositionPriceState.value } returns priceRange
```
Comment (276746) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:127]: I think that you cant do that and need to mockk the getter and setter instead to get the wanted behavior

Comment (ThomasPFMuller) [reply app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:127]: Good idea let me try

**Comment by ThomasPFMuller** [comment]
Resolves issue https://github.com/WanderWise-Inc/app/issues/172

File: app/src/main/java/com/github/wanderwise_inc/app/model/location/FakeItinerary.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/model/location/FakeItinerary.kt:1]: lgtm

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 2]
```diff
@@ -1,109 +1,80 @@
...
-      navHostController = mock(NavHostController::class.java)
-      firebaseAuth = mock(FirebaseAuth::class.java)
-      firebaseUser = mock(FirebaseUser::class.java)
-      `when`(firebaseAuth.currentUser).thenReturn(firebaseUser)
-      `when`(firebaseUser.uid).thenReturn(null)
+    every { mapViewModel.getItineraryFromUids(any()) } returns flow { emit(testItineraries) }
 
-      LikedScreen(mapViewModel, profileViewModel, navHostController, firebaseAuth)
+    composeTestRule.setContent {
+      FirebaseApp.initializeApp(LocalContext.current)
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:63]: good job abstracting the firebase construct

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 3]
```diff
@@ -112,9 +83,102 @@ class LikedScreenTest {
...
+
+    // no itineraries should be displayed
+    testExpectedItinerariesDisplayed(listOf(), textInput = "Tokyo")
+
+    // should only correspond to SF and Switzerland itineraries
+    testExpectedItinerariesDisplayed(listOf(FakeItinerary.SWITZERLAND), textInput = "Switzerland")
+  }
+
+  @Test
+  fun `price range filters itineraries correctly`() {
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:116]: relevent tests, good job!

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt [hunk 4]
```diff
@@ -112,9 +83,102 @@ class LikedScreenTest {
...
+      // composeTestRule.onRoot(useUnmergedTree = true).printToLog()
+      for (i in expectedItineraries.indices) {
+        composeTestRule
+            .onNodeWithTag(TestTags.ITINERARY_LIST_SCROLLABLE)
+            .performScrollToIndex(i) // scroll to correct position
+        composeTestRule
+            .onNodeWithTag("${TestTags.ITINERARY_BANNER}_${expectedItineraries[i].uid}")
+            .assertIsDisplayed()
+      }
+      for (itinerary in testItineraries.minus(expectedItineraries)) {
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:177]: looks good to me.

File: app/src/test/java/com/github/wanderwise_inc/app/ui/ScrollableItineraryListTest.kt [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class ScrollableItineraryListTest {
```
Comment (YofiY) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/ScrollableItineraryListTest.kt:29]: good general use of mokito.

**Review by YofiY — APPROVED** [review]
Overall, the tests seem to be quite extensive and cover enough lines. Excellent job Thomas.

File: app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/LikedScreenTest.kt:1]: Good testing and clean code, well done

File: app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt [hunk 6]
```diff
@@ -102,9 +81,110 @@ class OverviewScreenTest {
...
+
+    // SF and Switzerland itineraries should be displayed
+    testExpectedItinerariesDisplayed(listOf(FakeItinerary.SWITZERLAND, FakeItinerary.SAN_FRANCISCO))
+
+    testExpectedItinerariesDisplayed(listOf(FakeItinerary.SWITZERLAND), timeRange = 5f..12f)
+
+    testExpectedItinerariesDisplayed(listOf(FakeItinerary.SAN_FRANCISCO), timeRange = 1f..3.5f)
+  }
+
+  private fun testExpectedItinerariesDisplayed(
```
Comment (engjellismaili) [inline app/src/test/java/com/github/wanderwise_inc/app/ui/OverviewScreenTest.kt:142]: is it a test function ? if yes then why isn't a @Test tag ?

**Review by engjellismaili — COMMENTED** [review]
Overall clean testing code, well done.

**Pull request merged**